src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -543,7 +545,12 @@ static void rr_connectivity_changed_locked(grpc_exec_ctx *exec_ctx, void *arg,
   }
   // If the policy is shutting down, unref and return.
   if (p->shutdown) {
+    const bool needs_shutdown = !sd->subchannel_list->shutting_down;
     rr_subchannel_list_unref(exec_ctx, sd->subchannel_list, "pol_shutdown");
+    if (needs_shutdown) {
+      rr_subchannel_list_shutdown(exec_ctx, sd->subchannel_list,
```

Comment:
Is it safe to call shutdown after calling unref?  What if unref causes the subchannel list to be destroyed?